File: app/src/main/java/org/thunderdog/challegram/service/PushHandler.java
package org.thunderdog.challegram.service;

import android.app.Service;

import org.drinkless.tdlib.TdApi;
import org.jetbrains.annotations.NotNull;
import org.thunderdog.challegram.TDLib;
import org.thunderdog.challegram.telegram.TdlibManager;
import org.thunderdog.challegram.tool.UI;
import org.thunderdog.challegram.unsorted.Settings;

import java.util.Map;

import tgx.bridge.PushReceiver;
import tgx.td.JSON;

public final class PushHandler implements PushReceiver {
  @Override
  public void onNewToken (@NotNull Service service, @NotNull TdApi.DeviceToken token) {
    UI.initApp(service.getApplicationContext());
    TDLib.Tag.notifications("onNewToken %s, sending to all accounts", token);
    TdlibManager.instance().runWithWakeLock(manager -> {
      manager.setDeviceToken(token);
    });
  }

  @Override
  public void onMessageReceived (@NotNull Service service, @NotNull Map<@NotNull String, ?> message, long sentTime, int ttl) {
    UI.initApp(service.getApplicationContext());
    final long pushId = Settings.instance().newPushId();

    String payload = JSON.stringify(message);

    PushProcessor pushProcessor = new PushProcessor(service);
    pushProcessor.processPush(pushId, payload, sentTime, ttl);
  }
}

File: app/src/main/java/org/thunderdog/challegram/service/PushHandler.kt
package org.thunderdog.challegram.service

import android.app.Service
import org.drinkless.tdlib.TdApi
import org.thunderdog.challegram.Log
import org.thunderdog.challegram.TDLib
import org.thunderdog.challegram.telegram.TdlibManager
import org.thunderdog.challegram.tool.UI
import org.thunderdog.challegram.unsorted.Settings
import tgx.bridge.PushManager
import tgx.td.stringify

class PushHandler : PushManager {
  override fun onNewToken(service: Service, token: TdApi.DeviceToken) {
    UI.initApp(service.applicationContext)
    log("onNewToken %s, sending to all accounts", token)
    TdlibManager.instance().runWithWakeLock { manager ->
      manager.setDeviceToken(token)
    }
  }

  override fun onMessageReceived(service: Service, message: Map<String, Any>, sentTime: Long, ttl: Int) {
    UI.initApp(service.applicationContext)
    val pushId = Settings.instance().newPushId()

    val payload = stringify(message)

    val pushProcessor = PushProcessor(service)
    pushProcessor.processPush(pushId, payload, sentTime, ttl)
  }

  override fun log(format: String, vararg args: Any) =
    TDLib.Tag.notifications(format, args)

  override fun error(message: String, error: Throwable?) {
    TDLib.Tag.notifications("$message: ${Log.toString(error)}")
  }
}